src/Microsoft.DotNet.VersionTools/Util/FileUtils.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.DotNet.VersionTools.Util
+{
+    static class FileUtils
+    {
+        public static bool ReplaceFileContents(string path, Func<string, string> replacement)
+        {
+            string contents;
+            Encoding encoding;
+
+            // Atttempt to preserve the file's encoding, using ASCII encoding (which has no BOM) if the file's
+            // encoding cannot be detected. 
+            using (StreamReader reader = new StreamReader(new FileStream(path, FileMode.Open), Encoding.ASCII, true))
```

Comment:
If you create the UTF8Encoding object yourself, you can create one that doesn't write a BOM.

`new UTF8Encoding(encoderShouldEmitUTF8Identifier: false)`

This is better because it handles unicode characters in the file.